packages/atoms/src/Button/IconButton.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { TouchableOpacity } from 'react-native';
+
+import { Icon, IconName } from '@trezor/icons';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { ButtonColorScheme, ButtonSize } from './Button';
+
+type Props = {
+    iconName: IconName;
+    onPress: () => void;
+    isRounded?: boolean;
+    colorScheme?: ButtonColorScheme;
+    size?: ButtonSize;
+};
+
+type StyleProps = {
+    isRounded: boolean;
+    size: ButtonSize;
+    colorScheme: ButtonColorScheme;
+};
+
+const iconButtonStyle = prepareNativeStyle<StyleProps>(
+    (utils, { isRounded, colorScheme, size }) => {
+        const colorSchemeStyles = {
+            primary: {
+                backgroundColor: utils.colors.green,
+                color: utils.colors.white,
+            },
+            gray: {
+                backgroundColor: utils.colors.gray300,
+                color: utils.colors.gray700,
+            },
+        };
+
+        const sizeStyles = {
+            small: {
+                width: 16,
+                height: 16,
+                borderRadius: isRounded ? utils.borders.radii.round : utils.borders.radii.small / 2,
+            },
+            medium: {
+                width: 32,
+                height: 32,
+                borderRadius: isRounded ? utils.borders.radii.round : utils.borders.radii.small,
+            },
+            large: {
+                width: 48,
+                height: 48,
+                borderRadius: isRounded ? utils.borders.radii.round : utils.borders.radii.small,
+            },
+        };
+        return {
+            justifyContent: 'center',
+            alignItems: 'center',
+            ...colorSchemeStyles[colorScheme],
+            ...sizeStyles[size],
+        };
+    },
+);
+
+export const IconButton = ({
+    iconName,
+    onPress,
+    colorScheme = 'primary',
+    size = 'medium',
+    isRounded = false,
```

Comment:
I think that's unwanted. Why do you think there is a need for style prop? Since this is a component composed of variants, there should either be a variant or another component used but it's unwanted to pass styles and maybe change bgColor this way because that's not what this component is made for, right?

Reply:
Yes, as I see it, the appearance defined by the rest of the props (size, round props, color scheme) is enough for this component. You are right ;).